.github/workflows/test.yml

```
@@ -0,0 +1,46 @@
+name: lint and test
+
+on:
+   push:
+      branches:
+         - main
+   pull_request:
+      branches:
+         - main
+         - dev
+
+jobs:
+   Setup:
+      runs-on: ubuntu-latest
+      steps:
+         - name: Checkout branch
+           uses: actions/checkout@v3
+
+         - name: Get yarn cache directory path
+           id: yarn-cache-dir-path
+           run: echo "::set-output name=dir::$(yarn cache dir)"
+
+         - name: Setup Node.js
+           uses: actions/setup-node@v3
+           with:
+              node-version: 16.x
+
+         - uses: actions/cache@v3
+           id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+           with:
+              path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+              key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+              restore-keys: |
+                 ${{ runner.os }}-yarn-
+
+         - name: Install dependencies
+           run: yarn install --frozen-lockfile

```

--frozen-lockfileは非推奨
```suggestion
           run: yarn install --immutable

```